### PR TITLE
GTK fixes and cross compiling

### PR DESCRIPTION
**Pull request by jD91mZM2**
This PR includes:

 - Fixed GTK+.
 - Allow anybody to enable GTK+, but on Windows it's on by default (currently using a hacky build.rs script)
 - Files for cross compiling.

How do you cross compile? Simple!

First, build the docker container:
```
docker build -t spectrl .
```
Secondly, `export PKG_CONFIG_ALLOW_CROSS=1`
Lastly, use [cross](https://github.com/japaric/cross) like normal to compile the thing.

### CODE REVIEW

**Review by Mnpn — APPROVED** [review]
Looks good to me.

File: src/main.rs [hunk 1]
```diff
@@ -83,33 +77,66 @@ fn inner_main() -> Result<(), Error> {
...
+                let rect = label.get_allocation();
+                println!("{:?}", rect);
+                println!("{:?}", rgb);
+
+                ctx.rectangle(0.0, 0.0, rect.width as f64, rect.height as f64);
+                ctx.set_source_rgb(rgb.red, rgb.green, rgb.blue);
+                ctx.fill();
+
+                ctx.set_source_rgb(1.0 - rgb.red, 1.0 - rgb.green, 1.0 - rgb.blue);
+                ctx.show_text("YEE");
```
Comment (Mnpn) [inline src/main.rs]: Although this "YEE"?

Comment (jD91mZM2) [reply src/main.rs]: Whoops. Well, the secret is out. My debug messages are the weirdest.

Comment (Mnpn) [reply src/main.rs]: kek!

**Review by Mnpn — APPROVED** [review]
Thanks for fixing that!

**Comment by Mnpn** [comment]
Thanks!

File: src/main.rs (hunk 1 above)
Comment (jD91mZM2) [reply src/main.rs]: Yes the rest of my debug messages are `kek`. And `lol`, `wat`, `wot`, `woot`, `hey`, `hi`, `1`, `2`, et.c.

**Pull request merged**